src/main/java/com/gizasystems/automationexercise/pages/CartPage.java

```diff
@@ -13,30 +13,81 @@ public class CartPage {
+
+    private final By secondProductTotal = By.xpath("//p[@class='cart_total_price'][normalize-space()='Rs. 400']");
+
     // Constructor
-    public CartPage(SHAFT.GUI.WebDriver driver){
+    public CartPage(SHAFT.GUI.WebDriver driver) {
         this.driver = driver;
     }
 
     //////////////////// Actions \\\\\\\\\\\\\\\\\\\\
 
     //Clicking using JS as fix for pipeline failure on safari (Click isn't happening even with ClickUsingJS Flag)
     @Step("Open Cart Page")
-    public CartPage openCart(){
+    public CartPage openCart() {
         driver.element().clickUsingJavascript(viewCart_a);
         return this;
     }
 
     //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
     @Step("Verify Cart Page is Loaded")
-    public CartPage verifyCartPageIsLoaded(){
+    public CartPage verifyCartPageIsLoaded() {
         driver.verifyThat().element(proceedToCheckout_btn).isVisible().perform();
         return this;
     }
 
     @Step("Validate on Product Added To Cart Page")
-    public CartPage verifyProductAddedToCart(String addedProductName){
+    public CartPage verifyProductAddedToCart(String addedProductName) {
         driver.assertThat().element(productName_h4).text().isEqualTo(addedProductName).perform();
         return this;
     }
+
+    public CartPage navigate() {
+        driver.browser().navigateToURL(url);
+        return this;
+    }
+
+    @Step("Validate both products are added to Cart")
+    public CartPage validateOnItemsAddedInCart(String firstItemName, String secondItemName) {
+        driver.verifyThat().element(firstProductName).text().isEqualTo(firstItemName).perform();
+        driver.verifyThat().element(secondProductName).text().isEqualTo(secondItemName).perform();
+        return this;
+    }
+
+    @Step("Validate on the price of the products")
+    public CartPage validateOnProductPrices(String firstItemPrice, String secondItemPrice) {
+        driver.verifyThat().element(firstProductPrice).text().isEqualTo(firstItemPrice).perform();
+        driver.verifyThat().element(secondProductPrice).text().isEqualTo(secondItemPrice).perform();
+        return this;
+    }
+
+    @Step("Validate on the Quantity of the products")
+    public CartPage validateOnProductQuantity(String firstItemQuantity, String secondItemQuantity) {
+        driver.verifyThat().element(firstProductQuantity).text().isEqualTo(firstItemQuantity).perform();
+        driver.verifyThat().element(secondProductQuantity).text().isEqualTo(secondItemQuantity).perform();
+        return this;
+    }
+
+    @Step("Validate on the Total Price of the Products")
+    public CartPage validateOnTotalPrice(String firstItemTotalPrice, String secondItemTotalPrice) {
+        driver.verifyThat().element(firstProductTotal).text().isEqualTo(firstItemTotalPrice).perform();
+        driver.verifyThat().element(secondProductTotal).text().isEqualTo(secondItemTotalPrice).perform();
+        return this;
+    }
```

Comment:
Those methods are not dynamic; What if we have only 1 item, or if we have more than 2 items?.
We need to make those methods dynamic, not only works with 2 items.